include/swift/Driver/Action.h

```diff
@@ -48,6 +48,7 @@ class Action {
     AutolinkExtractJob,
     REPLJob,
     LinkJob,
+    ArchiveJob,
```

Comment:
I would like to get @jrose-apple's opinion on naming these: `DyamicLinkJob` and `StaticLinkJob`

Reply:
Yeah, I think I like @compnerd's names better (sans typo).

Reply:
Doing it in a separate commit (but still the same PR) might be nice for reviewing purposes, but not necessary.